### PR TITLE
Sun notes api

### PR DESCRIPTION
**Pull request by Sundera96**

### CODE REVIEW

File: public/scripts/week-calendar.js [hunk 1]
```diff
@@ -1069,18 +1069,28 @@ function checkNotesValidations(form) {
     notes_tag_error.innerText = "tag cant be longer than 20 characters";
     form.tag.setCustomValidity("error");
   }
-  if (!form.tag.value.match(/^[a-zA-Z]+$/)) {
+  if (
+    typeof form.tag.value === "string" &&
+    form.tag.value.trim().length > 0 &&
+    !form.tag.value.match(/^[a-zA-Z0-9_]+$/)
```
Comment (sanjeet-jain) [inline public/scripts/week-calendar.js:1075]: the regex needs to be present everywhere tag is validated ( currently its missing in alot of places ) 
image

File: utils/utils.js [hunk 2]
```diff
@@ -48,23 +48,23 @@ const utils = {
...
       throw new Error(
-        `${inputName} can not have spaces and contains only letters`
-      );
-    }
-    if (
-      inputName === "title" &&
-      !input
-        .trim()
-        .toLowerCase()
-        .match(/^(?![\d])[\w\s]+$/gi)
```
Comment (sanjeet-jain) [inline utils/utils.js:62]: needs to removed from everywhere 
image

File: views/partials/modals/notesModal.handlebars [hunk 3]
```diff
@@ -46,8 +46,8 @@
             {{! tag }}
             <div class="mb-3">
               <label for="notes_tag" class="form-label">Tag</label>
-              <input required type="text" class="form-control" name="tag" id="notes_tag" placeholder="Tag" value=""
-                maxlength="20" pattern="^[a-zA-Z]+$" />
+              <input type="text" class="form-control" name="tag" id="notes_tag" placeholder="Tag" value=""
+                maxlength="20" pattern="(^$)|(^[a-zA-Z0-9_]+$)" />
```
Comment (sanjeet-jain) [inline views/partials/modals/notesModal.handlebars:50]: make pattern consistent everywhere

File: views/partials/modals/tasksModal.handlebars [hunk 4]
```diff
@@ -25,7 +25,7 @@
             <div class="mb-3">
               <label for="task_title" class="form-label">Title</label>
               <input required type="text" class="form-control" name="title" id="task_title" placeholder="Title" value=""
-                maxlength="100" pattern="^(?=.{1,100}$)(?![\d])[\w\s]+$" />
```
Comment (sanjeet-jain) [inline views/partials/modals/tasksModal.handlebars:28]: also needs to be removed from other modals and places as well

File: views/partials/modals/tasksModal.handlebars [hunk 5]
```diff
@@ -43,7 +43,7 @@
             <div class="mb-3">
               <label for="task_tag" class="form-label">Tag</label>
               <input required type="text" class="form-control" name="tag" id="task_tag" placeholder="Tag" value=""
-                maxlength="20" pattern="^[a-zA-Z]+$" />
+                maxlength="20" pattern="(^$)|(^[a-zA-Z0-9_]+$)" />
```
Comment (sanjeet-jain) [inline views/partials/modals/tasksModal.handlebars:46]: make pattern consistent for tag throughout the modals

**Review by sanjeet-jain — CHANGES_REQUESTED** [review]

File: routes/meetings.js [hunk 6]
```diff
@@ -102,10 +102,7 @@ router
       );
       return res.status(200).json({ userId: userId, meetingId: meetingId });
     } catch (e) {
-      if (e === "Meeting Details havent Changed") {
-        return res.status(400).json({ error: e.message });
-      }
-      return res.status(500).json({ error: e.message });
+      return res.status(400).json({ error: e.message });
```
Comment (sanjeet-jain) [inline routes/meetings.js:105]: this might cause internal errors to go back as 400

File: routes/meetings.js [hunk 7]
```diff
@@ -180,9 +177,7 @@ router
         repeatingCounterIncrement,
         repeatingIncrementBy
       );
-      return res
-        .status(200)
-        .json({ userId: userId, meetingId: newMeeting._id });
+      return res.status(200).json({ userId: userId });
```
Comment (sanjeet-jain) [inline routes/meetings.js:180]: undo this

File: routes/tasks.js [hunk 8]
```diff
@@ -90,22 +116,51 @@ router
...
+      } else {
+        taskPutData.textBody = "";
+      }
+      if (typeof dateAddedTo === "string" && dateAddedTo.trim().length > 0) {
+        utils.validateDate(dateAddedTo, "dateAddedTo");
+        taskPutData.dateAddedTo = dayjs(dateAddedTo.trim()).format(
+          "YYYY-MM-DDTHH:mm"
+        );
+      } else {
+        taskPutData.dateAddedTo = "";
```
Comment (sanjeet-jain) [inline routes/tasks.js:136]: this should be null or empty string ? could you explain if the data layer is expecting a string to decide what to store in the mongo collection ?

**Review by sanjeet-jain — CHANGES_REQUESTED** [review]

File: routes/meetings.js (hunk 7 above)
Comment (Sundera96) [reply routes/meetings.js:180]: It will cause issue if I undo that as newMeetings id can also be array of meeting ids

Comment (sanjeet-jain) [reply routes/meetings.js:180]: cool

**Review by sanjeet-jain — APPROVED** [review]

**Pull request merged**